### PR TITLE
Update relay status if device rejoins as a relay

### LINKED ISSUE #367: Write unit test to cover the scenario reported in issue #183
#183

### LINKED ISSUE #1566: Rejoining device as a relay, doesn't mark it relay.
### Describe the bug

If you connect the device as a normal device, but if you reconnect the same device as a relay, it doesn't update it's relay flag.

### How to Reproduce

Connect nexd. Disconnect and reconnect it using `nexd relay` , and check the web-ui (or debug log), it won't update the relay flag.

### Expected Behavior

If the device rejoin as a Relay, it should be marked as a Relay. If we don't want to allow any device to convert to relay, we should deny onboarding of the device if it comes back as a relay.

### Deployment Environment

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by vishnoianil**
Fixes #1566 
Fixes #367

### CODE REVIEW

**Comment by russellb** [comment]
Can you add test coverage? It should be a pretty small addition to `integration-tests/features/device-api.feature`

File: internal/models/device.go [hunk 1]
```diff
@@ -54,4 +54,5 @@ type UpdateDevice struct {
 	Hostname       string     `json:"hostname" example:"myhost"`
 	Endpoints      []Endpoint `json:"endpoints" gorm:"type:JSONB; serializer:json"`
 	Revision       *uint64    `json:"revision"`
+	Relay          bool       `json:"relay"`
```
Comment (chirino) [inline internal/models/device.go:57]: Can we make this a *bool so that updating the relay status is optional?

**Review by chirino — CHANGES_REQUESTED** [review]

**Comment by vishnoianil** [comment]
> Can you add test coverage? It should be a pretty small addition to `integration-tests/features/device-api.feature`

Done

File: internal/models/device.go (hunk 1 above)
Comment (vishnoianil) [reply internal/models/device.go]: Done

**Review by russellb — APPROVED** [review]

**Review by chirino — APPROVED** [review]
Lgtm

**Pull request merged**